include/oneapi/dpl/pstl/histogram_impl.h

```diff
@@ -0,0 +1,77 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_HISTOGRAM_IMPL_H
+#define _ONEDPL_HISTOGRAM_IMPL_H
+
+#include "histogram_extension_defs.h"
+#include "histogram_binhash_utils.h"
+#include "iterator_impl.h"
+
+#if _ONEDPL_HETERO_BACKEND
+#    include "hetero/histogram_impl_hetero.h"
+#endif
+
+namespace oneapi
+{
+namespace dpl
+{
+
+namespace __internal
+{
+
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _Size, typename _IdxHashFunc,
+          typename _RandomAccessIterator2>
+oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy>
+__pattern_histogram(_ExecutionPolicy&& exec, _RandomAccessIterator1 __first, _RandomAccessIterator1 __last,
+                    _Size __num_bins, _IdxHashFunc __func, _RandomAccessIterator2 __histogram_first)
+{
+    static_assert(sizeof(_Size) == 0 /*false*/,
+                  "Histogram API is currently unsupported for policies other than device execution policies");
+}
+
+} // namespace __internal
+
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _Size, typename _RandomAccessIterator2,
+          typename _ValueType>
+oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, _RandomAccessIterator2>
+histogram(_ExecutionPolicy&& exec, _RandomAccessIterator1 first, _RandomAccessIterator1 last, _Size num_bins,
+          _ValueType first_bin_min_val, _ValueType last_bin_max_val, _RandomAccessIterator2 histogram_first)
+{
+    __internal::__pattern_histogram(
```

Comment:
Is it better (and we try to do that everywhere in oneDPL) to use fully qualified names => `oneapi::dpl::__internal::__pattern_histogram`

Reply:
added